gammapy/time/lightcurve.py

```diff
@@ -112,3 +116,249 @@ def compute_fvar(self):
         fvar_err = sigxserr / (2 * fvar)
 
         return fvar, fvar_err
+
+
+class LightCurveFactory(object):
+    """
+    Class producing light curve
+
+    Parameters
+    ----------
+    spectral_extraction: `~gammapy.spectrum.SpectrumExtraction`
+       class storing statistics, IRF and event lists
+    """
+
+    def __init__(self, spectral_extraction):
+        self.obs_list = spectral_extraction.obs
+        self.obs_spec = spectral_extraction.observations
+        self.off_evt_list = self._get_off_evt_list(spectral_extraction)
+        self.on_evt_list = self._get_on_evt_list(spectral_extraction)
+
+        
+    def _get_off_evt_list(self, spectral_extraction):
+        """
+        Returns list of OFF events for each observations
+        """
+        off_evt_list = []
+        for bg in spectral_extraction.background:
+            off_evt_list.append(bg.off_events)
+        return off_evt_list
+
+    
+    def _get_on_evt_list(self, spectral_extraction):
+        """
+        Returns list of OFF events for each observations
+        """
+        on_region = spectral_extraction.target.on_region
+        on_evt_list = []
+        for obs in spectral_extraction.obs:
+            idx = on_region.contains(obs.events.radec)
+            on_evt_list.append(obs.events.select_row_subset(idx))
+            
+        return on_evt_list
+    
+            
+    def _get_time_intervals(self, t_start, t_stop, time_binning_type):
+        """
+        Compute time intervals needed for a light curve
+
+        Parameters
+        ----------
+        time_start: `~astropy.units.Quantity`
+            start time to integrate events
+        time_stop: `~astropy.units.Quantity`
+            start time to integrate events
+        time_binning_type: `str`
+            type of time binning (e.g. obs, minute, hour, night, day, month, user)
+        """
+
+        dt = None
+        n_dt = None
+        intervals = []
+        if time_binning_type == 'obs':
+            n_dt = len(self.obs_list)
+            for obs in self.obs_list:
+                # for now use time of first and last evt to define the interval
+                # intervals.append([obs.tstart, obs.tstop])
+                intervals.append([obs.events.time[0], obs.events.time[-1]])
+        else:
+            raise NotImplementedError
+
+        return intervals
+
+    
+    def light_curve(self, t_start, t_stop, t_binning_type,
```

Comment:
Like I said, I'd suggest to simplify the arguments of this `lightcurve` method to compute lightcurves, and only take user-defined time binning here.

Also, this method is waaaaaaaay too long. I'll try to make suggestions how to split it into parts that can be read and tested separately.